demos/common/stream_client/stream_client.py

```diff
@@ -132,18 +143,24 @@ def grab_frame(self):
     dropped_frames = 0
     frames = 0
     def callback(self, frame, i, timestamp, result, error):
+        if error is not None:
+            if self.benchmark:
+                self.dropped_frames += 1
+            if self.verbose:
+                print(error)
+        if i == None:
+            i = result.get_response().parameters["OVMS_MP_TIMESTAMP"].int64_param
+        if timestamp == None:
+            timestamp = result.get_response().parameters["OVMS_MP_TIMESTAMP"].int64_param
         frame = self.postprocess_callback(frame, result)
         self.pq.put((i, frame, timestamp))
-        if error is not None and self.verbose == True:
-            print(error)
+        self.req_q.get()
 
     def display(self):
         i = 0 
         while True:
-            if self.pq.empty():
-                continue
             entry = self.pq.get()
-            if (entry[0] == i and self.exact) or (entry[0] > i and self.exact is not True):
+            if (entry[0] == i and self.exact and self.streaming_api is not True) or (entry[0] > i and (self.exact is not True or self.streaming_api is True)):
```

Comment:
Rename i -> displayed_frame_id?
Unpack queue entry:
send_frame_id, received_frame, timestamp = self.pq.get()
Make ifology easier to understand:
(entry[0] == i and self.exact and self.streaming_api is not True)
(entry[0] > i and (self.exact is not True or self.streaming_api is True))